### PR TITLE
fix: component styles and sign up / onboarding dark mode

### PR DESCRIPTION
**Pull request by henrikklee**
Hi all, this PR fixes some minor issues I encountered when setting up Cal.com locally. Also added dark mode support to the onboarding.

## What does this PR do?

- Fix some styling issues (see below)
- Add full dark mode support to the (new) sign up form and the whole onboarding process

**1. Fix missing vertical divider with `addOnLeading`:**

Before:
AddOnLeadingBefore

After (now congruent with`addOnSuffix`:
AddOnLeadingAfter

**2. Fix `PasswordField` border getting brighter (in dark mode) when focused but not hovered**
This issue is related to https://github.com/calcom/cal.com/commit/4438ca2a6341dca876272ab5ba33ff8cfd95e830 . Back then the `PasswordField` input was wrapped with a `<div className="absolute">` and some custom selectors. This wrapping `<div>` is not needed anymore because the show/hide password eye icon is now added to the input via the `addOnSuffix` prop. Removing this wrapper resolves the issue.

Before:
PasswordBefore

After:
PasswordAfter

**3. Fix corners in new sign up form**

Before:
SignupCornersBefore

After:
SignupCornersAfter

**4. Fix primary button color in dark mode**
In most places this was already applied manually. This PR adds the dark mode styling to the component itself for design consistency.

Before:
PrimaryButtonBefore

After:
PrimaryButtonAfter

**5. Enable dark mode for the (new) sign up form and the complete onboarding process**

Sign Up Form:
SignupDarkMode

Getting Started:
GettingStarted1

GettingStarted2

GettingStarted4

GettingStarted5

**6. Fix styling of Apple Calendar Setup**

Before:
AppleBefore

After:
AppleAfter

After (Light mode):
AppleAfterLight

## Requirement/Documentation

- none

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] Chore (refactoring code, technical debt, workflow improvements)

## How should this be tested?

Visual check of the affected routes (in system dark mode), especially:

- /signup
- /getting-started/[[step]]
- /apps/apple-calendar/setup

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: packages/ui/components/button/Button.tsx [hunk 1]
```diff
@@ -48,9 +48,9 @@ export const buttonClasses = cva(
       },
       color: {
         primary:
-          "bg-brand-default hover:bg-brand-emphasis focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset focus-visible:ring-brand-default text-brand disabled:bg-brand-subtle disabled:text-brand-subtle disabled:opacity-40 disabled:hover:bg-brand-subtle disabled:hover:text-brand-default disabled:hover:opacity-40",
+          "bg-brand-default hover:bg-brand-emphasis focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-offset focus-visible:ring-brand-default text-brand disabled:bg-brand-subtle disabled:text-brand-subtle disabled:opacity-40 disabled:hover:bg-brand-subtle disabled:hover:text-brand-default disabled:hover:opacity-40 dark:bg-white dark:text-black",
         secondary:
```
Comment (sean-brydon) [inline packages/ui/components/button/Button.tsx:52]: These values should be correct already - without the dark: prefix

If they were not working in a specific area of the app (onboarding) the variables may not have been set

**Review by sean-brydon — DISMISSED** [review]
Wow this is a great addition! 

Dark mode looks awesome well done. I have one ask that we refrain from using `dark:` anywhere possible. 

We have tokens defined in `cal.com/apps/web/styles/globals.css` 

If they are not working on a specific page that is another issue. The rest of the app functions fine in darkmode. 

Great PR thanks bud! 

Edit: Looking at this more i think the issue you are seeing here is the cal brand colour is not set yet - due to that being a user defined thing in settings.  

We will need to set them in classNames 

Light Mode:
    "--cal-brand": "#111827",
    "--cal-brand-emphasis": "#101010",
    "--cal-brand-text": "white",

Dark Mode: 
  --cal-brand: white;
  --cal-brand-emphasis: #e1e1e1;
  --cal-brand-text: black;

File: apps/web/pages/signup.tsx [hunk 2]
```diff
@@ -355,7 +355,10 @@ export default function Signup({
                       router.push(GOOGLE_AUTH_URL);
                     }}>
                     <img
-                      className={classNames("text-emphasis  mr-2 h-5 w-5", premiumUsername && "opacity-50")}
+                      className={classNames(
+                        "text-emphasis mr-2  h-5 w-5 dark:invert",
```
Comment (sean-brydon) [inline apps/web/pages/signup.tsx:358]: Lets invert the google icon here too

**Comment by stabildev** [comment]
> Wow this is a great addition!
> 
> Dark mode looks awesome well done.

Thanks!

> I have one ask that we refrain from using `dark:` anywhere possible.
> 

I copied that approach from /login.tsx (changed over there as well now).

Can confirm that everything looks as intended. With one exception:

image

IMO bg-transparent would make more sense on the secondary (outline) button:

image

**Comment by stabildev** [comment]
Originally refactored the sign up page using AuthContainer like this

image

At the same time that https://github.com/calcom/cal.com/commit/8dabbbd09fb084ea8d62d9460d569e59f95a22ed (signup refactor) happened

**Comment by stabildev** [comment]
> All looks good to me @stabildev - great first contribution! I will try and figure out why linter is failing after these calls 🙏

Thanks @sean-brydon . There seems to be a mismatch with prettier and eslint regarding tailwind class sorting (especially those [custom_selector_classes].

**Comment by sean-brydon** [comment]
> > All looks good to me @stabildev - great first contribution! I will try and figure out why linter is failing after these calls 🙏
> 
> Thanks @sean-brydon . There seems to be a mismatch with prettier and eslint regarding tailwind class sorting (especially those [custom_selector_classes].

Yeah i know! never seen it before and can't get it to pass either :')

**Review by sean-brydon — DISMISSED** [review]
LGTM

**Review by sean-brydon — APPROVED** [review]
LGTM

**Pull request merged**